src/rules/database-migration-filename-format.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Module dependencies.
+ */
+
+import dayjs from 'dayjs';
+import dayjsCustomParseFormat from 'dayjs/plugin/customParseFormat.js';
+import path from 'node:path';
+
+dayjs.extend(dayjsCustomParseFormat);
+
+/**
+ * Constants.
+ */
+
+const filenameRegex = /^([0-9]{14})-([a-zA-Z0-9-]*)$/;
+
+/**
+ * `database-migration-filename-format` rule.
+ * - Validates that database migration file names are formatted correctly.
+ *
+ * @type {import('eslint').Rule.RuleModule}
+ */
+
+const formatDatabaseMigrationFilename = {
+  create(context) {
+    return {
+      Program(node) {
+        const { filename: filepath } = context;
+        const { name: filename } = path.parse(filepath);
+        const regexMatchResult = filename.match(filenameRegex);
+
+        if (!regexMatchResult) {
+          context.report({
+            data: { filepath },
+            messageId: 'invalidFormat',
+            node
+          });
+
+          return;
+        }
+
+        const [, dateTimeString] = regexMatchResult;
+        const dateTime = dayjs(dateTimeString, 'YYYYMMDDHHmmss', true);
+
+        if (!dateTime.isValid()) {
+          context.report({
+            data: { dateTimeString, filepath },
+            messageId: 'malFormedDateTime',
+            node
+          });
+
+          return;
+        }
+
+        if (dateTime.isAfter(dayjs())) {
```

Comment:
Do we want to be this strict? What about giving it a bit of a threshold, like 1h/1d or something?
(my point is regarding if someone needs to change it manually due to some clash, but I might be over-engineering it)